File: platforms/neoforge-1.21.1/src/main/java/heckerpowered/lethal/platform/services/NeoForgeClientPlatform.java
/*
 * SPDX-License-Identifier: MIT
 * Copyright (c) 2026 heckerpowered
 */

package heckerpowered.lethal.platform.services;

import heckerpowered.bridge.adapter.client.MinecraftAccess;
import heckerpowered.bridge.platform.services.ClientPlatform;
import heckerpowered.lethal.platform.interop.MixinInterop;
import net.minecraft.client.Minecraft;
import org.jetbrains.annotations.NotNull;

public final class NeoForgeClientPlatform implements ClientPlatform {
    @Override
    @NotNull
    public MinecraftAccess getMinecraft() {
        return MixinInterop.requireAccess(Minecraft.getInstance(), MinecraftAccess.class);
    }
}

File: platforms/neoforge-1.21.1/src/main/kotlin/heckerpowered/lethal/platform/services/NeoForgeClientPlatform.kt
/*
 * SPDX-License-Identifier: MIT
 * Copyright (c) 2026 heckerpowered
 */

package heckerpowered.lethal.platform.services

import heckerpowered.bridge.adapter.client.MinecraftAccess
import heckerpowered.bridge.platform.services.ClientPlatform
import heckerpowered.bridge.requireAccess
import net.minecraft.client.Minecraft

class NeoForgeClientPlatform : ClientPlatform {
    override val minecraft: MinecraftAccess
        get() = requireAccess<MinecraftAccess>(Minecraft.getInstance())
}
